services/debugger/uploader.go

```diff
@@ -137,9 +152,13 @@ func (uploader *Uploader) handleEvents() {
 	}
 }
 
-func (uploader *Uploader) flushEvents() {
+func (uploader *Uploader) flushEvents(ctx context.Context) {
 	for {
 		time.Sleep(uploader.batchTimeout)
+		select {
+		case <-ctx.Done():
+		case <-time.After(uploader.batchTimeout):
```

Comment:
time.After inside this loop can create memory leaks, maybe we should use a ticker instead. 🤔

Reply:
My bad actually, in this case, we should be good.

Reply:
I don't understand why nothing is done though. The context is done and we don't care and keep going. Same with batchTimeout which in this case looks more like a backoff interval rather than a timeout.

Reply:
Why was this resolved? Shouldn't we return on `ctx.Done()` instead of flushing? WDYT?
Second thing, shouldn't we rename `batchTimeout` to `flushInterval` or something? It is not being used as a timeout, right?

Reply:
1. In case of ctx.Done() we fist want the events stored to be flashed and then return to exit the loop.
2. I like this suggestion. I will leave the configuration parameter we use to input the value for this as is for now though `Debugger.batchTimeoutInS`.

Reply:
> 1. In case of ctx.Done() we fist want the events stored to be flashed and then return to exit the loop.

Can you explain why please? Are we relying on the uploader in a way that if we don't upload the data is lost? If not and the context is canceled then we must honour it and return. WDYT?

Reply:
@fracasula Why is important for you to exit without flushing existing data?

flushing the data is not important, but it helps with testing and in ensures more consistent behaviour.

Reply:
@lvrach I think that is the expectation in every Go codebase. You pass a context, the context is canceled, all operations are terminated as soon as possible. Plus, this is an OpenSource project which makes conventions/standards even more important. To make an example the opposite would be as bad as having a method called `SendSync` which sends asynchronously instead. It says one thing but does another.

In any case, why would you cancel a context and keep operations running instead? Do you see an advantage on doing so?